.github/workflows/ids-check.yml

```diff
@@ -0,0 +1,110 @@
+name: "Check LLVM ABI annotations"
+on:
+  pull_request:
+    branches:
+      - main
+      - 'users/**'
+  push:
+    branches:
+      - 'main'
+      - 'release/**'
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    if: github.repository_owner == 'llvm'
+    name: Check LLVM_ABI annotations with ids
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
+
+    steps:
+      # TODO: Use an image with a prebuilt idt.
+      - uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8 # v4.2.2
+        with:
+          repository: compnerd/ids
+          path: ${{ github.workspace }}/ids
+          ref: b3bf35dd13d7ff244a6a6d106fe58d0eedb5743e # main
+
+      - uses: actions/checkout@8e8c483db84b4bee98b60c0593521ed34d9990e8 # v4.2.2
+        with:
+          path: ${{ github.workspace }}/llvm-project
+          fetch-depth: 2
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@24d32ffd492484c1d75e0c0b894501ddb9d30d62 # v47.0.0
+        with:
+          separator: ","
+          skip_initial_fetch: true
+          base_sha: 'HEAD~1'
+          sha: 'HEAD'
+
+      # TODO: Use an image with a prebuilt idt.
+      - name: Install dependencies
+        run: |
+          sudo apt install -y clang-19 ninja-build libclang-19-dev
+          pip install lit PyGithub
```

Comment:
These need to at least be version pinned, but ideally should be installed from a lockfile with hashes.